Runtime/Scripts/PersistentConnection.cs

```diff
@@ -131,192 +217,193 @@ IEnumerator OnIntervalEdgeEventsLocation(EdgeEventsConnection connection)
         {
-          if (config.latencyConfig.maxNumberOfUpdates > 0)
-          {
-            if (latencyUpdatesCounter >= config.latencyConfig.maxNumberOfUpdates)
+          latencyUpdatesRunning = false;
+          Logger.Log("Stopping Latency Updates according to configs, No. LatencyUpdates: " + latencyUpdatesCounter);
+          latencyUpdatesCounter = 0;
+          yield break;
+        }
+      }
+      yield return new WaitForSecondsRealtime(config.latencyConfig.updateIntervalSeconds);
+      yield return StartCoroutine(LocationService.EnsureLocation());
+      if (hasTCPPorts)
+      {
+        connection.TestConnectAndPostLatencyUpdate(host, (uint)config.latencyTestPort, location).ConfigureAwait(false);
+      }
+      else
+      {
+        integration.matchingEngine.EdgeEventsConnection
+          .TestPingAndPostLatencyUpdate
+          (integration.GetHost(), location).ConfigureAwait(false);
+      }
+      latencyUpdatesCounter++;
+      yield return StartCoroutine(OnIntervalEdgeEventsLatency(connection, host));
+    }
+
+    async void HandleReceivedEvents(ServerEdgeEvent edgeEvent)
+    {
+      Logger.Log("Received event type: " + edgeEvent.EventType);
+      FindCloudletEventTrigger trigger = event_trigger_dict[edgeEvent.EventType];
+      if (config.newFindCloudletEventTriggers.Contains(trigger))
+      {
+        switch (trigger)
+        {
+          case FindCloudletEventTrigger.Error:
+            PropagateError(FindCloudletEventTrigger.Error, edgeEvent.ErrorMsg);
+            return;
+          case FindCloudletEventTrigger.LatencyTooHigh:
+            ProcessLatency(edgeEvent.Statistics);
+            break;
+          default:
+            if (edgeEvent.NewCloudlet != null)
             {
-              latencyUpdatesRunning = false;
-              yield break;
+              PropagateSuccess(trigger, edgeEvent.NewCloudlet);
             }
-          }
-          yield return new WaitForSecondsRealtime(config.latencyConfig.updateIntervalSeconds);
-          yield return StartCoroutine(LocationService.EnsureLocation());
+            return;
+        }
+      }
+      else
+      {
+        if (edgeEvent.EventType == ServerEventType.EventLatencyRequest)
+        {
+          string host = integration.GetHost();
+          Loc location = LocationService.RetrieveLocation();
           if (hasTCPPorts)
           {
-            connection.TestConnectAndPostLatencyUpdate(host, (uint)config.latencyTestPort, location).ConfigureAwait(false);
+            await integration.matchingEngine.EdgeEventsConnection.TestConnectAndPostLatencyUpdate(host, (uint)config.latencyTestPort, location);
           }
           else
           {
-            integration.matchingEngine.EdgeEventsConnection
-              .TestPingAndPostLatencyUpdate
-              (integration.GetHost(), location).ConfigureAwait(false);
+            await integration.matchingEngine.EdgeEventsConnection.TestPingAndPostLatencyUpdate(host, location);
           }
```

Comment:
I don't think we have to do anything if config.newFindCloudletEventTriggers does not contain the trigger. EventLatencyRequest is most likely sent by the developer and they are also setting the FIndCloudletTriggers. So, if "EventLatencyRequest" is not set here, then we should assume that they don't want latency tests at all

Reply:
Fixed to Ignore LatencyRequest if LatencyTooHigh is not a trigger

Reply:
hmm i'm confused. it looks like the code from line 321 to line 333 is still there. that can all be removed

Reply:
Will need to fix that.